src/main/java/edu/harvard/iq/dataverse/util/SignpostingResources.java

```diff
@@ -0,0 +1,337 @@
+                    .add("type", Json.createArrayBuilder()
+                            .add(jsonObjectBuilder().add("href", "https://schema.org/AboutPage"))
+                            .add(jsonObjectBuilder().add("href", defaultFileTypeValue))
+                    );
+        } else {
+            mandatory = jsonObjectBuilder()
+                    .add("anchor", landingPage)
+                    .add("cite-as", Json.createArrayBuilder().add(jsonObjectBuilder().add("href", ds.getPersistentURL())))
+                    .add("type", Json.createArrayBuilder()
+                            .add(jsonObjectBuilder().add("href", "https://schema.org/AboutPage"))
+                    );
+        }
+
+        if (authors != null) {
+            mandatory.add("author", authors);
+        }
+        if (licenseString != null && !Objects.equals(licenseString, "")) {
+            mandatory.add("license", jsonObjectBuilder().add("href", licenseString));
+        }
+        if (!mediaTypes.toString().trim().isEmpty()) {
+            mandatory.add("describedby", mediaTypes);
+        }
+        if (items != null) {
+            mandatory.add("item", items);
+        }
+        linksetJsonObj.add(mandatory);
+
+        // remove scholarly type as shown already on landing page
+        for (FileMetadata fm : fms) {
+            DataFile df = fm.getDataFile();
+            JsonObjectBuilder itemAnchor = jsonObjectBuilder().add("anchor", getPublicDownloadUrl(df));
+            itemAnchor.add("collection", Json.createArrayBuilder().add(jsonObjectBuilder()
+                    .add("href", landingPage)));
+            linksetJsonObj.add(itemAnchor);
+        }
+
+        return linksetJsonObj;
+    }
+
+
+    private String getPublicDownloadUrl(DataFile dataFile) {
+        StorageIO<DataFile> storageIO = null;
+        try {
+            storageIO = dataFile.getStorageIO();
+        } catch (IOException e) {
+            logger.warning(String.format("Error getting storageID from file; original error message is: %s", e.getLocalizedMessage()));
+        }
+
+        if (storageIO instanceof SwiftAccessIO) {
+            String fileDownloadUrl;
+            SwiftAccessIO<DataFile> swiftIO = (SwiftAccessIO<DataFile>) storageIO;
+            try {
+                swiftIO.open();
+            } catch (IOException e) {
+                logger.warning(String.format("Error opening the swiftIO; original error message is: %s", e.getLocalizedMessage()));
+            }
+
+            //if its a public install, lets just give users the permanent URL!
+            if (systemConfig.isPublicInstall()) {
+                fileDownloadUrl = swiftIO.getRemoteUrl();
+            } else {
+                //TODO: if a user has access to this file, they should be given the swift url
+                // perhaps even we could use this as the "private url"
+                fileDownloadUrl = swiftIO.getTemporarySwiftUrl();
```

Comment:
This would be temporary, i.e. access would expire. In the file page, where someone is navigating now to get the file, this works but as a permanent link I think this is problematic.

Reply:
I think URLs like http://localhost:8080/api/access/datafile/15 would be best practice. I don't think you get that from the swiftIO.getTemporarySwiftUrl() call. That download URL - /api/access/datafile - comes from the api/Access.java class where I think you could add a header. (Similarly, I think you could respond to a HEAD instead of GET there as well).